src/compiler/reverse.jl

```diff
@@ -3,14 +3,17 @@ using IRTools: IR, Variable, Pipe, xcall, var, prewalk, postwalk,
   insertafter!, finish, expand!, prune!, substitute!, substitute,
   block, block!, branch!, return!, stmt, meta
 
+# va = var args? replaces the last nothing/AbstractZero by N Zero()s
 @inline tuple_va(N, xs) = xs
 @inline tuple_va(N, x, xs...) = (x, tuple_va(N, xs...)...)
-@inline tuple_va(::Val{N}, ::Nothing) where N = ntuple(_ -> nothing, Val(N))
+@inline tuple_va(::Val{N}, ::Nothing) where N = ntuple(_ -> Zero(), Val(N))
+@inline tuple_va(::Val{N}, ::AbstractZero) where N = ntuple(_ -> Zero(), Val(N))
```

Comment:
```suggestion
@inline tuple_va(::Val{N}, x::AbstractZero) where N = ntuple(_ -> x, Val(N))
```
So that Zero and DoesNotExist stay the same as before